### DISCUSSION TITLE
Обход блокировок VPN

### DISCUSSION THREAD

**Discussion by Gleb-pro-admin** [bol-van/zapret#463] [Ideas]
Здесь обсуждаем обход блокировок легкодетектируемых протоколов, таких как `OpenVPN` (TCP и UDP), Wireguard, IPsec ESP, PPtP и других

**Comment by Dr0rby** [comment]
@bol-van Добрый день. Видел вы сделали скрипт на обход блокировки wireguard. Все отлично работает. Спасибо.
Вопрос: Можете сделать точно такой же обход, но для OpenVPN и/или IKEv2?
Или может подскажите, как правильно сделать?

**Reply by bol-van** [reply]
Как обычно в этом деле есть 2 шага. Перенаправление трафика и стратегия дурения.
Перенаправлять можно или по содержимому пакета, по номерам портов или по начальному количеству пакетов (connbytes только в linux).
Дурить неизвестные протоколы стандартным образом. --dpi-desync-any-protocol и ограничитель connbytes или cutoff. Без ограничителя если есть фейки в стратегии будет трэш флуд

**Reply by Dr0rby** [reply]
@bol-van А как понять что именно надо перенаправлять?

В вашем скрипте, как я понял, это отвечает за перенаправление?

```
zapret_custom_firewall()
{
        # $1 - 1 - run, 0 - stop

        local f='-p udp -m u32 --u32'
        fw_nfqws_post $1 "$f 0>>22&0x3C@4>>16=0x9c&&0>>22&0x3C@8=0x01000000" "$f 44>>16=0x9c&&48=0x01000000" $QNUM_WG4ALL
}

**Reply by SilentWarrior1** [reply]
> @bol-van Добрый день. Видел вы сделали скрипт на обход блокировки wireguard. Все отлично работает. Спасибо. Вопрос: Можете сделать точно такой же обход, но для OpenVPN и/или IKEv2? Или может подскажите, как правильно сделать?

доброго дня, а можете меня пнуть на тот скриптик? в win10-cистеме работать будет?

**Reply by Dr0rby** [reply]
> > @bol-van Добрый день. Видел вы сделали скрипт на обход блокировки wireguard. Все отлично работает. Спасибо. Вопрос: Можете сделать точно такой же обход, но для OpenVPN и/или IKEv2? Или может подскажите, как правильно сделать?
> 
> доброго дня, а можете меня пнenm на тот скриптик? в win10-cистеме работать будет?

https://github.com/bol-van/zapret/blob/master/init.d/custom.d.examples.linux/50-wg4all

Скрипт для zapret на роутерах.
Не знаю как его под Винду сделать

**Reply by SilentWarrior1** [reply]
> Скрипт для zapret на роутерах. Не знаю как его под Винду сделать

благодарю, может что ув. bol-van подскажет для винды

**Reply by bol-van** [reply]
На винде можно написать собственный --wf-raw фильтр для windivert на базе сохраненного через --wf-save стандартного.
Но это не имеет большого смысла. Достаточно сделать --wf-udp на порт WG и написать для него стратегию.
Если WG разделяет порт с другими сервисами, например он сидит на udp 443, то можно дополнительно применять --filter-l7=wireguard

**Reply by SilentWarrior1** [reply]
>Достаточно сделать --wf-udp на порт WG и написать для него стратегию. 

Если вас не затруднит, можете подсказать пару-тройку стратегий для WG? Знаний в этой области пока немного, но самообучаюсь (пытаюсь), очень позитивно, когда вижу наглядные рабочие (понятное дело, что всё индивидуально) примеры, можно от них отталкиваться в таком случае.

**Reply by bol-van** [reply]
Для udp немного вариантов.   fake , возможно с увеличенным количество repeats
ipfrag2 только для своего сервера и только на внешнем IP адресе
или через ipv6

**Reply by spvkgn** [reply]
> Если вас не затруднит, можете подсказать пару-тройку стратегий для WG?

Под линуксом хватает дефолтного fake с нулями с добавлением repeats не меньше 4. Хотя, на разных провайдерах по-разному. Возможно, кастомный fake вместо дефолтных нулей пройдет без повторов, надо проверять.

**Reply by SilentWarrior1** [reply]
> Для udp немного вариантов. fake , возможно с увеличенным количество repeats ipfrag2 только для своего сервера и только на внешнем IP адресе или через ipv6

у меня один из платных впнов несколько дней назад перестал соединяться (без использования встроенной обфускации по SS) по любым протоколам/портам, а если использую первоначально подключённый Netch (ну или v2rayN), то через 853 udp соединяет, но траффик не идёт, и соединение отваливается. Раньше хотя бы так работало (после успешного подключения к впн, netch отключал.)

пытался составить кофиг, наверное (да и скорее всего) для вас он будет вида "тяп-ляп", но надо же с чего-то начинать...

```
--wf-l3=ipv4 --wf-tcp=80,443 --wf-udp=443,853,37000-38000,50000-50099 --dpi-desync=split2 --dpi-desync-split-seqovl=49 --dpi-desync-split-pos=50 --wssize 1:6 --new
--filter-udp=443,853,37000-38000,50000-50099 --dpi-desync=fake --dpi-desync-cutoff=d2 --dpi-desync-repeats=6 --dpi-desync-any-protocol --new
--filter-l7=wireguard --dpi-desync=fake --dpi-desync-repeats=6 --dpi-desync-fake-wireguard="C:\YTDisBystro\fake\quic_initial_www_google_com.bin"
```
edited: опробовал - не помогает, скорее всего, либо я напортачил где-то, либо... провайдер слишком злой

**Reply by bol-van** [reply]
Последний профиль никогда не сработает, потому что весь udp забирает тот, что перед ним

**Reply by Fenrir-0xFF** [reply]
> Хотя, на разных провайдерах по-разному.
> Возможно, кастомный fake вместо дефолтных нулей пройдет без повторов, надо проверять.

Нули использовать не стоит, dpi первый байт анализирует, по крайней мере на билайне точно, но даже несмотря на это количество повторов fake удалось сократить минимум до двух, меньше двух уже не прокатывает.

**Reply by spvkgn** [reply]
> Нули использовать не стоит, dpi первый байт анализирует, по крайней мере на билайне точно, но даже несмотря на это количество повторов fake удалось сократить минимум до двух, меньше двух уже не прокатывает.

У меня без разницы - и с нулями и с рандомными 64 байтами меньше 4ех повторов не срабатывает

**Reply by Fenrir-0xFF** [reply]
> У меня без разницы - и с нулями и с рандомными 64 байтами меньше 4ех повторов не срабатывает

Попробуй syndata использовать, правда ХЗ как он себя с UDP ведёт, Я его только на TCP тестил, байты в нём --dpi-desync-fake-syndata= тоже роль играют причём серьёзнее чем на фейке, у меня нули не подошли а рандомные как повезёт - некоторые прокатывают а некоторые нет, причём в syndata DPI проверяет не только первый байт, однако удалось сократить количество байт до одного и прокатил байт 0x0A

**Reply by bol-van** [reply]
SYN относится к TCP. На UDP его нет
